### DISCUSSION TITLE
Increasing File Size

### DISCUSSION THREAD

**Discussion by Schrank97** [ImageMagick/ImageMagick#2964] [Development]
Hello,

I've been using ImageMagick for a few projects now, and I somehow... got confused about one thing. I'm creating a new Image via the RenderTargetBitmap and decode it to a MemoryStream with a TiffBitmapEncoder. So far so good.

With this Stream, I've got the Bytes i need to write a file. So I... did this. I've checked the result via PhotoShop, and it was quite good. Quality was perfect, Pixels were perfect, and the file had following details:

File Size: 2MB
- Pixels:   Width 2126, Height 1346
dpi: 600 (both, Vert and Horz)
Bit Depth: 32
Compression Level: LZW

Color Space: **RGBA**

Exactly the image i wanted to have, except one thing: For my project, I need a CMYKa-Image. So here, ImageMagick comes in.
I used the same bytes from the Stream to create a Instance of MagickImage, change it to cmyk, and save it again. 
 
MagickImage mgImg = new MagickImage(ImageStream.ToArray());
            mgImg.ColorSpace = ColorSpace.CMYK;            
            mgImg.Write(ResultTIFFCMYK);

When i check the file now, it has the increased size of 13,6 MB and loses it's compression.

So i tried to just load the Stream into the MagickImage again, and saved it without changing to cmyk, simply as RGBa again.
It also loses it's compression and has the size of 10 MB now.

I opened all three Images in Photoshop, but you can't hardly see a difference between the images.

By debugging the process, the MagickImage's property "Compression" is set as "LZW",  but when saving it to the file, it loses it.

1. Question: Why is it not getting saved with the right compression Level? Even if i write the following line before saving, the Compression won't be saved to the file:

mgImg.SetCompression(CompressionMethod.LZW);

2. Question: Is the increased size only because of the compression, or does it have other causes? I see, that the Bitdepth of cmyka is 40, while rgba is 32. But i can't understand, why only 1 Byte per Pixel can make that increasing file size.

Thank you for reading & every answer !

**Comment by snibgo** [comment]
What version of IM?

You don't say what file format you use. From "mgImg.Write(ResultTIFFCMYK)" I guess it is TIFF. Is that correct?

TIFF CMYK compression works fine. For example:
```
f:\web\im>%IMG7%magick -size 2126x1346 xc: -colorspace CMYK -depth 8 x.tiff

f:\web\im>dir x.tiff

10/12/2020  12:41        11,446,662 x.tiff

f:\web\im>%IMG7%magick -size 2126x1346 xc: -colorspace CMYK -depth 8 -compress LZW x.tiff

f:\web\im>dir x.tiff
10/12/2020  12:41             9,636 x.tiff
```

As a general point, converting from sRGB to CMYK using "-colorspace CMYK" is a simple and fast arithmetic process. But for best quality, it is better to use ICC profiles. This would make no difference to pixel compression, but would need extra space for the embedded profile.

**Reply by Schrank97** [reply]
Hey Snibgo, thank you for your quick response.

I'm using the Version 7.21.1.0 of the .Net DLL.
And yes, you guessed right. It's all made with .Tiff formats.

In your example, i see, that the compression works fine. But I'm still wondering, why it doesn't work, when i try to apply it to my Image. Maybe I can find an other option to achieve that goal. Thank you for your answer anyways ! :)

Best Regards

**Reply by dlemstra** [reply]
You should use `image.Compression = CompressionMethod.LZW` instead of calling `image.SetCompression`.
